test/e2e/toolchain_test.go

```diff
@@ -65,41 +69,77 @@ func TestToolchain(t *testing.T) {
 			HasSpec(cheSub.Spec)
 	})
 
-	t.Run("should create subscription for tekton with TektonInstallation", func(t *testing.T) {
+	t.Run("should recreate che operator's ns operatorgroup subscription when ns deleted", func(t *testing.T) {
+		// given
+		ns := &v1.Namespace{}
+		err := f.Client.Get(context.TODO(), types.NamespacedName{Name: cheOperatorNs}, ns)
+		require.NoError(t, err)
+
 		// when
-		err := f.Client.Create(context.TODO(), tektonInstallation, cleanupOptions(ctx))
+		err = f.Client.Delete(context.TODO(), ns)
 
 		// then
-		require.NoError(t, err, "failed to create toolchain TektonInstallation")
+		require.NoError(t, err, "failed to delete Che Operator Namespace")
 
-		err = await.WaitForTektonInstallConditions(tektonInstallation.Name, wait.UntilHasTektonStatusCondition(tekton.SubscriptionCreated()))
+		err = await.WaitForNamespace(cheOperatorNs)
 		require.NoError(t, err)
 
-		AssertThatSubscription(t, tektonSub.Namespace, tektonSub.Name, f.Client).
+		err = await.WaitForCheInstallConditions(cheInstallation.Name, wait.UntilHasCheStatusCondition(che.SubscriptionCreated()))
+		require.NoError(t, err)
+
+		AssertThatNamespace(t, cheOperatorNs, f.Client).
 			Exists().
-			HasSpec(tektonSub.Spec)
+			HasLabels(toolchain.Labels())
+
+		AssertThatOperatorGroup(t, cheOg.Namespace, cheOg.Name, f.Client).
+			Exists().
+			HasSize(1).
+			HasSpec(cheOg.Spec)
+
+		AssertThatSubscription(t, cheSub.Namespace, cheSub.Name, f.Client).
+			Exists().
+			HasSpec(cheSub.Spec)
```

Comment:
I believe that these assertions could be extracted to separated function that would be called at the end of every subtest.

Reply:
Done https://github.com/codeready-toolchain/toolchain-operator/pull/14/commits/5f1514aa59b436829a7b4b4537b892aa33afeef5